src/models.py

```diff
@@ -0,0 +1,439 @@
+                     SUM(value) as value
+                   FROM event
+                     INNER JOIN measure USING(measure_uri)
+                   WHERE 1=1 ''' + clause + '''
+                   GROUP BY year, measure_uri, namespace, source, type, version
+                   ORDER BY year, measure_uri;'''
+            result = db.query(q, params)
+            return result
+        except (Exception, psycopg2.DatabaseError) as error:
+            logger.error(error)
+            raise Error(FATAL)
+
+    @staticmethod
+    def aggregate_by_measure_month(clause, params):
+        try:
+            q = '''SELECT
+                     measure_uri,
+                     namespace,
+                     source,
+                     type,
+                     version,
+                     to_char(timestamp, 'MM') as month,
+                     SUM(value) as value
+                   FROM event
+                     INNER JOIN measure USING(measure_uri)
+                   WHERE 1=1 ''' + clause + '''
+                   GROUP BY measure_uri, namespace, source, type, version,
+                     month
+                   ORDER BY measure_uri, month;'''
+            result = db.query(q, params)
+            return result
+        except (Exception, psycopg2.DatabaseError) as error:
+            logger.error(error)
+            raise Error(FATAL)
+
+    @staticmethod
+    def aggregate_by_month_measure(clause, params):
+        try:
+            q = '''SELECT
+                     to_char(timestamp, 'MM') as month,
+                     measure_uri,
+                     namespace,
+                     source,
+                     type,
+                     version,
+                     SUM(value) as value
+                   FROM event
+                     INNER JOIN measure USING(measure_uri)
+                   WHERE 1=1 ''' + clause + '''
+                   GROUP BY month, measure_uri, namespace, source, type,
+                     version
+                   ORDER BY month, measure_uri;'''
+            result = db.query(q, params)
+            return result
+        except (Exception, psycopg2.DatabaseError) as error:
+            logger.error(error)
+            raise Error(FATAL)
+
+
+class Measure(object):
+    def __init__(self, measure_uri, namespace, source, mtype, version):
+        self.measure_uri = measure_uri
+        self.namespace = namespace
+        self.source = source
+        self.type = mtype
+        self.version = version
+
+    def load_description(self):
+        self.description = self.get_description().list()
+
+    def get_description(self):
+        options = dict(uri=self.measure_uri)
+        q = '''SELECT locale_code, locale_name, description
+               FROM measure_description INNER JOIN locale USING(locale_code)
+               WHERE measure_uri = $uri
+               ORDER BY locale_code;'''
+        description = db.query(q, options)
+        return description if description else []
```

Comment:
If `[]` is returned, the `self.description = self.get_description().list()` call above will crash with the error:  `AttributeError: 'list' object has no attribute 'list'`.

I can't tell if this would get caught anywhere, but it may be worth adding a try/except or debug since `result_to_measure` does get called in quite a few different places.

Reply:
you're absolutely right - it now simply returns description and then the `load_description()` method defaults to an empty list